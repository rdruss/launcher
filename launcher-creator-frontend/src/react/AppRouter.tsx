import * as React from 'react';
import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom';
import LoginPageContainer from './pages/LoginPageContainer';
import HomePageContainer from './pages/HomePageContainer';

class AppRouterProps {
  public readonly authenticated: boolean;
}

const AppRouter = ({ authenticated }: AppRouterProps) => {
  if (!authenticated) {
    return (
      <BrowserRouter>
        <Route path="/">
          <Switch>
            <Route path="/login" component={LoginPageContainer} />
            <Redirect from="/" to="/login" />
          </Switch>
        </Route>
      </BrowserRouter>
    );
  }
  return (
    <BrowserRouter>
      <Route path="/">
        <Switch>
          <Route exact={true} path="/" component={HomePageContainer} />
          <Redirect from="/" to="/" />
        </Switch>
      </Route>
    </BrowserRouter>
  );
};

export default AppRouter;