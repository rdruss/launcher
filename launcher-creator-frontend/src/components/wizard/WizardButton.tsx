import * as React from 'react';
import * as Patternfly from 'patternfly-react';

interface LizardButtonProps {
  type: 'next' | 'launch';
  title?: string;
  onClick?: () => void;
  disabled?: boolean;
}

const WizardButton: React.StatelessComponent<LizardButtonProps> = ({ type, title, onClick, disabled = false }) => {
  let text = 'Launch';
  let icon = 'check';
  if (type === 'next') {
    text = 'Next';
    icon = 'angle-right';
  }
  if (title) {
    text = title;
  }
  return (
    <Patternfly.Button className={'wizard-button'} onClick={onClick} disabled={disabled}>
      <Patternfly.Icon type="fa" name={icon} /> {text}
    </Patternfly.Button>
  );
}

export default WizardButton;