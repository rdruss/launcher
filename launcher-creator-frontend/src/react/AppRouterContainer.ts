import { connect } from 'react-redux';
import { AppState } from '../redux/states';
import AppRouter from './AppRouter';

const mapStateToProps = (state: AppState) => ({
  authenticated: state.authentication.authenticated,
});

const mapDispatchToProps = (dispatch) => ({
});

const AppRouterContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(AppRouter);

export default AppRouterContainer;
