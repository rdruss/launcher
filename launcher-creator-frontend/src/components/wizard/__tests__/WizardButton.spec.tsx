import * as React from 'react';
import { shallow } from 'enzyme';

import WizardButton from '../WizardButton';

describe('<WizardButton />', () => {
  it('renders the button with type next', () => {
    const component = shallow(<WizardButton type={'next'} />);
    expect(component.find('Button').contains('Next')).toBeTruthy();
    expect(component.find('Button Icon').prop('name')).toBe('angle-right');
    expect(component).toMatchSnapshot();
  });

  it('renders the button with type launch', () => {
    const component = shallow(<WizardButton type={'launch'} />);
    expect(component.find('Button').contains('Launch')).toBeTruthy();
    expect(component.find('Button Icon').prop('name')).toBe('check');
    expect(component).toMatchSnapshot();
  });

  it('renders the button disabled', () => {
    const component = shallow(<WizardButton type={'launch'} disabled={true} />);
    expect(component.find('Button').prop('disabled')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('call function when clicking', () => {
    const mockCallBack = jest.fn();
    const component = shallow(<WizardButton type={'launch'} onClick={mockCallBack} />);
    component.find('Button').simulate('click');
    expect(mockCallBack).toHaveBeenCalled();
  });
});

