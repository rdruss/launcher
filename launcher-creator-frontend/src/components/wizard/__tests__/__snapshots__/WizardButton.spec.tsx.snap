// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WizardButton /> renders the button disabled 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  className="wizard-button"
  disabled={true}
>
  <Icon
    name="check"
    type="fa"
  />
   
  Launch
</Button>
`;

exports[`<WizardButton /> renders the button with type launch 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  className="wizard-button"
  disabled={false}
>
  <Icon
    name="check"
    type="fa"
  />
   
  Launch
</Button>
`;

exports[`<WizardButton /> renders the button with type next 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  className="wizard-button"
  disabled={false}
>
  <Icon
    name="angle-right"
    type="fa"
  />
   
  Next
</Button>
`;
